auditbeat/module/file_integrity/_meta/fields.yml

```diff
@@ -2,6 +2,319 @@
   title: File Integrity
   description: These are the fields generated by the file_integrity module.
   fields:
+  - name: file
+    type: group
+    description: File attributes.
+    fields:
```

Comment:
What's the delta in the list from ECS? For the non-ECS fields maybe we should [capitalize](https://www.elastic.co/guide/en/ecs/current/ecs-custom-fields-in-ecs.html#_capitalization) them to avoid the potential for future conflicts. Or propose them as additions to ECS.

Reply:
The difference is essentially adding fields from one OS that don't exist in another plus some totally new fields. So this is the change.

Existing:

- `file.elf.sections`
- `file.elf.sections.name`
- `file.elf.sections.physical_size`
- `file.elf.sections.virtual_size`
- `file.elf.sections.entropy`
- `file.elf.imports`
- `file.pe.imphash`

The following are added because they are new analyses from the parsers added here.

ELF:

- `file.elf.sections.var_entropy`
- `file.elf.imports_names_entropy`
- `file.elf.imports_names_var_entropy`
- `file.elf.go_import_hash`
- `file.elf.go_imports`
- `file.elf.go_imports_names_entropy`
- `file.elf.go_imports_names_var_entropy`
- `file.elf.go_stripped`
- `file.elf.import_hash` — alias for ELF equivalent of imphash to simplify cross-platform searches

PE:

- `file.pe.sections.var_entropy`
- `file.pe.imports_names_entropy`
- `file.pe.imports_names_var_entropy`
- `file.pe.go_import_hash`
- `file.pe.go_imports`
- `file.pe.go_imports_names_entropy`
- `file.pe.go_imports_names_var_entropy`
- `file.pe.go_stripped`

The following are added to PE as analogous to the existing ELF fields. All bar virtual_size are strictly informative (`virtual_size` is always equal to `physical_size` since only ELF has compressed sections)

- `file.pe.sections`
- `file.pe.sections.name`
- `file.pe.sections.physical_size`
- `file.pe.sections.virtual_size` — always the same as `physical_size` on PE (meaningful on ELF as a potential marker for evasive ELF)
- `file.pe.sections.entropy`
- `file.pe.import_hash` — alias for imphash to simplify cross-platform searches
- `file.pe.imports`

ECS doesn't have any fields for Mach-O so all the following are new:

- `file.macho.sections`
- `file.macho.sections.name`
- `file.macho.sections.physical_size`
- `file.macho.sections.virtual_size`
- `file.macho.sections.entropy`
- `file.macho.sections.var_entropy`
- `file.macho.import_hash` — alias for Mach-O equivalent of imphash
- `file.macho.symhash` — alias for Mach-O equivalent of for imphash
- `file.macho.imports`
- `file.macho.imports_names_entropy`
- `file.macho.imports_names_var_entropy`
- `file.macho.go_import_hash`
- `file.macho.go_imports`
- `file.macho.go_imports_names_entropy`
- `file.macho.go_imports_names_var_entropy`
- `file.macho.go_stripped`

The fields in ECS for this kind of information are not orthogonally defined even though in ~all cases here they could be, so I'm not sure what should be done.

Reply:
@ebeahan Do you have an opinion on what should be done here?

Reply:
@dcode @peasead contributed some of these fields to ECS as well as far more expertise in this area than I do. 

@dcode @peasead would you be able to weigh into the discussion here?

Reply:
Is there a reason we wouldn't formally add these to ECS?

Reply:
Added